File: tests/exercises/time-2-code/solution/solution.py
# -------------------------
# Subprograms
# -------------------------
# Load a single line of data
def load(filename):
    file = open(filename, "r")
    user = file.read()
    file.close()
    user = user.strip()
    return user


# Save a single line of data
def save(user, filename):
    file = open(filename, "w")
    user = user + "\n"
    file.write(user)
    file.close()


# -------------------------
# Main program
# -------------------------
user = load("datafile.txt")
# If the file is empty, ask for the data to save...
if user == "":
    print("Hello, I don't believe we have met.")
    user = input("What is your name? ")
    save(user, "datafile.txt")
    print("Nice to meet you", user + ".")
# ...otherwise display the data in the file
else:
    print("It's good to see you again,", user + ".")

File: tests/exercises/time-2-code/solution/solution.js
const fs = require('fs');
const readline = require('readline');

function load(filename) {
    try {
        return fs.readFileSync(filename, 'utf8').trim();
    } catch (err) {
        return "";
    }
}

function save(user, filename) {
    fs.writeFileSync(filename, `${user}\n`, 'utf8');
}

const user = load("datafile.txt");

if (user === "") {
    console.log("Hello, I don't believe we have met.");
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    // Empty string to wait silently for input
    rl.question("", (name) => {
        save(name, "datafile.txt");
        console.log(`Nice to meet you ${name}.`);
        rl.close();
    });
} else {
    console.log(`It's good to see you again, ${user}.`);
}
